### ISSUE TITLE
List of models that can be used as a backbone

### ISSUE THREAD

**Issue by bamblebam** [PyTorchLightning/lightning-flash#718]
## 🚀 Feature

There is no central list of models that can be used as a backbone in the docs. It becomes difficult to search for which model can be used for what task.  
Here in the [docs](https://lightning-flash.readthedocs.io/en/latest/api/generated/flash.text.seq2seq.translation.model.TranslationTask.html#flash.text.seq2seq.translation.model.TranslationTask) the info for the backbone isn't clear either. And it is the case in a lot of other places as well.  
Also in the README, in the first code example for `TranslationTask` the weights are loaded from a checkpoint using an AWS url. How does one find that url and model name?
### Motivation
This is really annoying when you are trying to build a quick model for any task and you have to browse through flash as well as huggingface or any other libraries docs just to check if the model will work.
This might be similar to #593 

### Pitch
Add a central list or table separated by task that contains the name of all the backbones or models that can be used for that particular task.

**Comment by akihironitta** [comment]
@bamblebam I agree. There could probably be a single page listing all available backbones for all tasks as you suggest in addition to available pretrained models #593.

**Comment by ethanwharris** [comment]
Hi @bamblebam we're looking in to ways to have this info in our docs. In the meantime, it's worth noting (and we should definitely document this as well) that you can do `Task.available_backbones` or (where relevant) `Task.available_heads`. E.g. for the `ImageClassifier` you would do:
```
print(ImageClassifier.available_backbones)
```

Hope that helps :smiley:

**Comment by bamblebam** [comment]
Yeah thanks for that. If you need help with the docs let me know I could make a PR.

**Comment by ethanwharris** [comment]
@bamblebam Definitely could use the help! Would be awesome to have a page in the docs about how to use the `available_backbones`, `available_heads` etc. methods as a starting point :smiley:

**Comment by bamblebam** [comment]
So you want a new page or should I just add doc strings for the available_backbones and available_heads methods. Because I checked the file and most of the methods didn't have doc strings.
if you want a new page in what folder of docs should I create the file and what should I name it ?

**Comment by ethanwharris** [comment]
Let's have both docstrings and a little guide (can be separate PRs if you like :smiley:). For the guide, could you add a page in `docs/source/general`, don't mind what you call it. Then be sure to add a reference to it in `docs/source/index.rst`, but we can help with stuff like that :smiley:

**Comment by bamblebam** [comment]
I wanted to ask how can I build the docs to check whether everything looks proper or not ?

**Comment by ethanwharris** [comment]
Hey @bamblebam to build the docs you should be able to do the following:
```
pip install -r requirements/docs.txt
cd docs
make html
```

You should then be able to view the docs with `_build/html/index.html`.

If that doesn't work, you can also open a PR and have the docs build there (can be viewed by clicking on the `details` button next to `ci/circleci: build-Docs artifact`).

Hope that helps :smiley:

**Comment by bamblebam** [comment]
Thanks that works.

**Comment by bamblebam** [comment]
Hey I created a draft PR take a look whether this is how it should look or not. If yes I will add the other models as well.

This is how this looks.

![image](https://user-images.githubusercontent.com/53450613/132251157-e6155fb5-464b-46f9-8e91-54bce4a8ef38.png)

**Comment by aniketmaurya** [comment]
In my last Pie & AI meetup, I talked about backbones and heads. This is the [video](https://youtu.be/72b1SbY9Ht0?t=656) 😄
